app/controllers/categories_controller.rb

```diff
@@ -1,6 +1,13 @@
 class CategoriesController < ApplicationController
   before_action :logged_in_user
   def index
-    @categories = Category.paginate(page: params[:page])
+    if params[:filter].eql?("learned")
+      @categories = current_user.categories.where.not(lessons: {result: nil})
+    elsif params[:filter].eql?("unlearned")
+      @categories = Category.left_outer_joins(:lessons).where.not(id: current_user.categories.where.not(lessons: {result: nil})).group(:id)
```

Comment:
@CharlotteDS 
You can make this line more readable, by separating methods to lines, like this:
![image](https://user-images.githubusercontent.com/20507565/50734457-7087c700-11da-11e9-8b11-613ed06a9994.png)